### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by andriy-shymkiv**

### CODE REVIEW

File: src/makeCalculator.js [hunk 1]
```diff
@@ -37,7 +37,51 @@
...
+      callback(number);
+
+      return this;
+    },
+
+    add(num) {
+      calculator.result += num;
+
+      return this;
+    },
```
Comment (Svitlana-Yudina) [inline src/makeCalculator.js:59]: You don't need return here, I think)

**Review by Svitlana-Yudina — APPROVED** [review]
It's a good solution as for me) 🔥

File: src/makeCalculator.js (hunk 1 above)
Comment (Andriy-Shymkiv) [reply src/makeCalculator.js:59]: I will try to remove return, but i'm not sure if it will work properly)

Comment (illiarerun) [reply src/makeCalculator.js:59]: You don't call these methods explicitly, so indeed you don't need it

File: src/makeCalculator.js [hunk 2]
```diff
@@ -37,7 +37,51 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    reset() {
+      calculator.result = 0;
```
Comment (illiarerun) [inline src/makeCalculator.js:44]: right now we can execute this code:
image
i guess it should work a bit other way:
- we should change object that we called `operate` **on**,
- operation shouldn't be dependant on exact object that you call it from
- you can somehow pass needed object from operate to this actual `operation`, try `this`

File: src/makeCalculator.js [hunk 3]
```diff
@@ -37,7 +37,51 @@
...
+      calculator.result += num;
+
+      return this;
+    },
+
+    subtract(num) {
+      calculator.result -= num;
+
+      return this;
+    },
```
Comment (illiarerun) [inline src/makeCalculator.js:65]: make this functions like a [pure ones](https://www.geeksforgeeks.org/pure-functions-in-javascript/), pass your result as first param, and number as a second param. do it so your operations won't be dependant on previous result

**Review by illiarerun — CHANGES_REQUESTED** [review]

File: src/makeCalculator.js [hunk 4]
```diff
@@ -37,7 +37,43 @@
...
+
+    multiply(num1, num2) {
+      return num1 * num2;
+    },
+
+    divide(num1, num2) {
+      if (num1 === 0) {
+        throw Error(`We can't divide by zero`);
+      }
```
Comment (illiarerun) [inline src/makeCalculator.js:70]: we should check divisor (дільник), not the divident (ділене)

**Review by illiarerun — CHANGES_REQUESTED** [review]

File: src/makeCalculator.js [hunk 5]
```diff
@@ -37,7 +37,47 @@
...
+      return num1 / num2;
+    },
+  };
+
+  return calculator;
 }
 
+const calculatorre = makeCalculator();
+
+console.log(calculatorre.operate(calculatorre.add, 10).operate(calculatorre.divide, 0));
```
Comment (mykola-koval) [inline src/makeCalculator.js:81]: remove logs

**Review by mykola-koval — CHANGES_REQUESTED** [review]

**Review by illiarerun — APPROVED** [review]

**Pull request closed without merging**